magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/KMSDiscovery.java

```diff
@@ -107,27 +105,32 @@ private void discoverAliases(KmsClient client, KeyListEntry resource, ObjectNode
   }
 
   private void discoverKeyPolicies(KmsClient client, KeyListEntry resource, ObjectNode data, Logger logger, ObjectMapper mapper) {
-    logger.info("Getting KeyPolicies for {}", resource.keyArn());
+    logger.info("Getting keyPolicies for {}", resource.keyArn());
     final String keyname = "keyPolicies";
     getAwsResponse(
-      () -> client.listKeyPolicies(ListKeyPoliciesRequest.builder().keyId(resource.keyId()).build()),
+      () -> client.listKeyPoliciesPaginator(ListKeyPoliciesRequest.builder().keyId(resource.keyId()).build()).policyNames()
+        .stream()
+        .collect(Collectors.toList()),
       (resp) -> AWSUtils.update(data, Map.of(keyname, resp)),
       (noresp) -> AWSUtils.update(data, Map.of(keyname, noresp))
     );
   }
 
   private void discoverGrants(KmsClient client, KeyListEntry resource, ObjectNode data, Logger logger, ObjectMapper mapper) {
-    logger.info("Getting Grants for {}", resource.keyArn());
+    logger.info("Getting grants for {}", resource.keyArn());
     final String keyname = "grants";
     getAwsResponse(
-      () -> client.listGrants(ListGrantsRequest.builder().keyId(resource.keyId()).build()),
+      () -> client.listGrantsPaginator(ListGrantsRequest.builder().keyId(resource.keyId()).build()).grants()
+        .stream()
+        .map(r -> r.toBuilder())
+        .collect(Collectors.toList()),
```

Comment:
After fix discovering grants increase KMS  discovery time from 45s to 3m30s.
Do we still want to discover it?

Reply:
That's interesting.  Perhaps we'll make it a configurable option in `AWSDiscoveryConfig`